internal/storage/sbom_store_test.go

```diff
@@ -0,0 +1,348 @@
+			Name:      "test",
+			Namespace: "default",
+		},
+	}
+
+	key := keyPrefix + "/default/test"
+
+	tests := []struct {
+		name             string
+		preconditions    *storage.Preconditions
+		validateDeletion storage.ValidateObjectFunc
+		expectedError    error
+	}{
+		{
+			name:          "happy path",
+			preconditions: &storage.Preconditions{},
+			validateDeletion: func(_ context.Context, _ runtime.Object) error {
+				return nil
+			},
+			expectedError: nil,
+		},
+		{
+			name:          "deletion fails with incorrect UID precondition",
+			preconditions: &storage.Preconditions{UID: ptr.To(types.UID("incorrect-uid"))},
+			validateDeletion: func(_ context.Context, _ runtime.Object) error {
+				return nil
+			},
+			expectedError: storage.NewInvalidObjError(key, "Precondition failed: UID in precondition: incorrect-uid, UID in object meta: "),
+		},
+	}
+
+	for _, test := range tests {
+		suite.Run(test.name, func() {
+			err := suite.store.Create(context.Background(), key, sbom, &v1alpha1.SBOM{}, 0)
+			suite.Require().NoError(err)
+
+			out := &v1alpha1.SBOM{}
+			err = suite.store.Delete(context.Background(), key, out, test.preconditions, test.validateDeletion, nil)
+
+			if test.expectedError != nil {
+				suite.Require().Error(err)
+				suite.Equal(test.expectedError.Error(), err.Error())
+			} else {
+				suite.Require().NoError(err)
+				suite.Equal(sbom, out)
+
+				err = suite.store.Get(context.Background(), key, storage.GetOptions{}, &v1alpha1.SBOM{})
+				suite.True(storage.IsNotFound(err))
+			}
+		})
+	}
+}
+
+func (suite *sbomStoreTestSuite) TestWatchEmptyResourceVersion() {
+	key := keyPrefix + "/default/test"
+	opts := storage.ListOptions{ResourceVersion: ""}
+
+	watcher, err := suite.store.Watch(context.Background(), key, opts)
+	suite.Require().NoError(err)
+
+	suite.broadcaster.Shutdown()
+
+	events := collectEvents(watcher)
+	suite.Require().Empty(events)
+}
+
+func (suite *sbomStoreTestSuite) TestWatchResourceVersionZero() {
+	key := keyPrefix + "/default/test"
+	sbom := &v1alpha1.SBOM{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "default",
+		},
+	}
+	err := suite.store.Create(context.Background(), key, sbom, &v1alpha1.SBOM{}, 0)
+	suite.Require().NoError(err)
+
+	opts := storage.ListOptions{ResourceVersion: "0"}
+
+	watcher, err := suite.store.Watch(context.Background(), key, opts)
+	suite.Require().NoError(err)
+
+	validateDeletion := func(_ context.Context, _ runtime.Object) error {
+		return nil
+	}
+	err = suite.store.Delete(context.Background(), key, &v1alpha1.SBOM{}, &storage.Preconditions{}, validateDeletion, nil)
+	suite.Require().NoError(err)
+
+	suite.broadcaster.Shutdown()
+
+	events := collectEvents(watcher)
+	suite.Require().Len(events, 2)
+	suite.Equal(watch.Added, events[0].Type)
+	suite.Equal(sbom, events[0].Object)
+	suite.Equal(watch.Deleted, events[1].Type)
+	suite.Equal(sbom, events[1].Object)
+}
+
+func (suite *sbomStoreTestSuite) TestWatchSpecificResourceVersion() {
```

Comment:
Not related with this test, just picking a "random" line. I think we should also test watches that are using selectors